### PR TITLE
CXF-88930, CXF-86483 Added Terratest in Prod and Also Added GHA for Prod Port connections

### PR DESCRIPTION
**Pull request by jkallem-equinix**
- Port2Alibaba
- Port2IBM2
- Port2Port
- Port2PrivateSP
- Port2AWS
- Port2Azure

### CODE REVIEW

File: .github/workflows/terratests-prod-suite.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: './go.mod'
+        id: go
+
+      - name: Get dependencies
+        run: |
+              go mod download
+      - uses: hashicorp/setup-terraform@v3
```
Comment (srushti-patl) [inline .github/workflows/terratests-prod-suite.yaml:36]: Add space between get dependencies and hashicorp/setup-terraform@v3 steps

File: .github/workflows/terratests-prod-suite.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          echo $TEST_DATA_PROD_PORT_2_AZURE_CONNECTION >> "./tests/examples-without-external-providers/port-2-azure-connection/terraform.tfvars.json"
+          echo $TEST_DATA_PROD_PORT_2_IBM2_CONNECTION >> "./examples/port-2-ibm2-connection/terraform.tfvars.json"
+          echo $TEST_DATA_PROD_PORT_2_PORT_CONNECTION >> "./examples/port-2-port-connection/terraform.tfvars.json"
+          echo $TEST_DATA_PROD_PORT_2_PRIVATE_SERVICE_PROFILE_CONNECTION >> "./examples/port-2-private-service-profile-connection/terraform.tfvars.json"
+
+      - name: Run Go Tests
+        run:
+          go test ./tests/prod -v -coverprofile coverage_prod_modules.txt -covermode=atomic -count 1 -parallel 8 -run "(DIGP)" -timeout 180m
+
+      - name: Setup Node.js 20.11.1
```
Comment (srushti-patl) [inline .github/workflows/terratests-prod-suite.yaml:54]: Remove Setup Node.js 20.11.1 step from the workflow

**Comment by jkallem-equinix** [comment]
Screenshot 2024-04-09 at 4 22 18 PM
Screenshot attached

**Review by srushti-patl — APPROVED** [review]
LGTM

**Pull request merged**